ddtrace/compat.py

```diff
@@ -103,7 +103,10 @@ def process_time_ns():
 
 
 if sys.version_info.major < 3:
-    main_thread = threading._MainThread()
+    if isinstance(threading.current_thread(), threading._MainThread):
+        main_thread = threading.current_thread()
+    else:
+        main_thread = threading._MainThread()
```

Comment:
@jd I'm not 100% about still instantiating a `_MainThread` here. I added a test for when the library is loaded in a thread and there are no obvious problems. It doesn't look like there's any way to get a reference to the main thread or its ident so we might not have much of a choice

Reply:
```suggestion
        main_thread = threading._shutdown.im_self
```
Just do this for Python 2. It's utterly hackish, but… who cares at this point, it's not going to break any time soon 😆